#170947435 Build sign-in feature

RusimbiPatrick opened the pull request:
### What does this PR do?

- Have sign-in feature build
- Have sign-in feature tested

### How should this be manually tested?

- Clone the [repo](https://github.com/andela/octopus-bn-backend.git)
- Run ``` npm install ```
-  Hit `http://localhost:3000/api/v1/auth/signin` to test it.

### Implementation plan

[Implementation plan name](https://drive.google.com/drive/u/1/folders/10O6cku3mfx2EQjTnghkYTr4tixe8xe-A)

### Stories associated with PR

[##170947435](https://www.pivotaltracker.com/story/show/170947435)

### Screenshots if any
<img width="1120" alt="Screen Shot 2020-02-12 at 13 10 11" src="https://user-images.githubusercontent.com/54870069/74329779-1c65f380-4d99-11ea-88e0-9a2a1736c2b5.png">
]
**On a different type of user**
<img width="1123" alt="Screen Shot 2020-02-12 at 12 01 32" src="https://user-images.githubusercontent.com/54870069/74324793-3fd87080-4d90-11ea-8e0b-fb7511d4c25f.png">

RusimbiPatrick committed:
feat(sign-in):create sign-in feature

- Create signin service
- Create signin controller
- Create sign-in route
- edit files to fix db configuration bugs

[Maintains #170947435]

kcharles52 commented on src/controllers/user.controller.js:
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable require-jsdoc */
+import service from '../services/user.services';
```
```suggestion
import service from '../services/userServices';
```

kcharles52 commented on src/database/migrations/20200206093213-create-users.js:
```diff
@@ -28,4 +40,4 @@ export default {
   }),
   down: (queryInterface) => queryInterface.dropTable('Users')
-};
+};
```
```suggestion
};

```

kcharles52 commented on src/routes/api/user.routes.js:
```diff
@@ -0,0 +1,8 @@
+import express from 'express';
+import userController from '../../controllers/user.controller';
```
```suggestion
import userController from '../../controllers/userController';
```

kcharles52 commented on src/routes/api/user.routes.js:
```diff
@@ -0,0 +1,8 @@
+const router = express.Router();
+
+
+router.post('/signin', userController.signin);
```
```suggestion
router.post('/signin', userController.signin);

```

kcharles52 commented on src/routes/api/user.routes.js:
```diff
@@ -0,0 +1,8 @@
+
+const router = express.Router();
+
+
```
```suggestion
```

kcharles52 commented on src/services/user.services.js:
```diff
@@ -0,0 +1,65 @@
+/* eslint-disable require-jsdoc */
```
```suggestion

```

kcharles52 commented on src/services/user.services.js:
```diff
@@ -0,0 +1,65 @@
+// import serialize from '../utils/jwtTokenizer';
+
+
+class userService {
```
add documentation for the class and methods

kcharles52 commented on src/services/user.services.js:
```diff
@@ -0,0 +1,65 @@
+      }
+
+      const { method } = registered;
+      if (method !== 'local') {
```
```suggestion
      
      if (method !== 'local') {
```

kcharles52 commented on src/test/signin.test.js:
```diff
@@ -0,0 +1,63 @@
+      });
+  });
+
+  it('It should show an error if the uses uses Open Auth', (done) => {
```
check the grammar used in the test description

blaise82 committed:
feat(user): social login with facebook and google (#13)

- create user routes
- create passport config
- create user controller

[Finishes #170947436]
